TypeCobol.Test/Parser/Programs/Cobol85/RenamesAndUsage.Mix.txt

```diff
@@ -0,0 +1,505 @@
+      *   "PICTURE X(1)"was assumed.
+       01 Group-C01 group-usage is national.
+Line 381[15,26] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+           05 Array-C01-KO.
+           66 Array-C01-OK-R renames Array-C01-KO .
+      
+      
+      *TODO Wait for support of UTF-8 (#2504)
+      *01 Group-C02 group-usage is UTF-8.
+      *    05 Array-C02-KO.
+      *    66 Array-C02-KO-R renames Array-C02-KO .
+      
+      
+      *********************************************
+      *D. Usage is not declared directly on parent (test recursivity)
+      *********************************************
+      
+       01 Group-D1 usage is pointer.
+           05 Group-D11.
+                10 Array-D1-OK.
+                66 Array-D1-OK-R renames Array-D1-OK .
+      
+      
+      
+       01 Group-D2 comp-5.
+           05 Group-D21.
+      
+      *The"PICTURE"clause for item"ARRAY-D2-KO"was not compatible with
+      *the specified"USAGE"."USAGE DISPLAY"was assumed.
+      *A"PICTURE"clause was not found for elementary item"ARRAY-D2-KO".
+      *   "PICTURE X(1)"was assumed.
+Line 409[20,30] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+                10 Array-D2-KO.
+                66 Array-D2-OK-R renames Array-D2-KO .
+      
+      
+      
+      
+       01 Group-D3 usage is procedure-pointer.
+           05 Group-D31.
+                10 Group-D311.
+                    15 Array-D3-OK.
+                    66 Array-D3-OK-R renames Array-D3-OK .
+      
+      
+       01 Group-D4 usage is display-1.
+           05 Group-D41.
+                10 Group-D411.
+      
+      *The"PICTURE"clause for item"ARRAY-D4-KO"was not compatible with
+      *the specified"USAGE"."USAGE DISPLAY"was assumed.
+      *A"PICTURE"clause was not found for elementary item"ARRAY-D4-KO".
+      *   "PICTURE X(1)"was assumed.
+Line 430[24,34] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+                    15 Array-D4-KO.
+                    66 Array-D4-OK-R renames Array-D4-KO .
+      
+      
+      
+      
+      
+       01 Group-D5 usage is comp-1.
+           05 Array-D5-OK.
+      *The specified "USAGE" was different from the "USAGE" specified at
+      *the group level.  The group "USAGE" was assumed for this item.
+           05 Group-D51 group-usage is national.
+      
+                10 Group-D511.
+      *Ok because group-usage is national is not considered valid by IBM
+      * compiler. Usage comp-1 from the first parent with an usage is
+      *considered in this case.
+Line 447[24,34] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+                    15 Array-D5-KO.
+                    66 Array-D5-OK-R renames Array-D5-KO .
```

Comment:
Same remark as on previous file, I'd like to have a comment explainig the behavior of the parser.

Reply:
Comment changed. Is it better?